### PR TITLE
Stackrox release 4.1

### PR DESCRIPTION
**Pull request by ghost**
CI configuration files to support Stackrox release `4.1`.

### CODE REVIEW

**Comment by jschnath** [comment]
Paging @gavin-stackrox and @vjwilson for approval.I just went over the pull request checklist as per the upstream release instructions but did not find anything wrong with the PR.

**Review by vjwilson — APPROVED** [review]

**Comment by jschnath** [comment]
Paging @gavin-stackrox and @vjwilson again. It looks like after making a change this needs approval to be tested again.

Previously the sharding was failing and telling me that there are `Config file does not belong to any auto-updating config` which presumably was because they had 4.1 in their name which did not match the pattern. I changed it to match 4.* now in my latest commit. 

Also is there a different process for changing the `prow/02_config/_plugins.yaml` file? Since now it is asking me to get additional approval from DPTP which I think is the test platform team. Perhaps I'm doing something wrong.

File: core-services/prow/02_config/_plugins.yaml [hunk 1]
```diff
@@ -1201,12 +1201,12 @@ config_updater:
         - ci
       gzip: true
       name: ci-operator-2.x-configs
-    ci-operator/config/stackrox/stackrox/*-release-4.0*.yaml:
+    ci-operator/config/stackrox/stackrox/*-release-4.*.yaml:
```
Comment (vikin91) [inline core-services/prow/02_config/_plugins.yaml:1204]:
```suggestion
    ci-operator/config/stackrox/stackrox/*-release-4.1*.yaml:
```

File: core-services/prow/02_config/_plugins.yaml [hunk 2]
```diff
@@ -1201,12 +1201,12 @@ config_updater:
...
       clusters:
         app.ci:
         - ci
       gzip: true
-      name: ci-operator-4.0-configs
+      name: ci-operator-4.x-configs
```
Comment (vikin91) [inline core-services/prow/02_config/_plugins.yaml:1209]:
```suggestion
      name: ci-operator-4.1-configs
```

**Review by vikin91 — COMMENTED** [review]
Could you maybe try this?

Those regexps are matching too much, maybe making them more specific would help?

**Comment by gavin-stackrox** [comment]
I think the change to `_plugins.yaml` without the `__ocp` _should_ work. I cannot see that exact change in the commit history. 
For some reference, see the comment from @msugakov [here](https://github.com/openshift/release/pull/37753#discussion_r1154216193) relating to the glob match for that config.

File: core-services/prow/02_config/_plugins.yaml [hunk 3]
```diff
@@ -2233,6 +2239,12 @@ config_updater:
         - ci
       gzip: true
       name: job-config-4.0
+    ci-operator/jobs/stackrox/stackrox/*-release-4.1__ocp*.yaml:
+      clusters:
+        app.ci:
+          - ci
+      gzip: true
+      name: ci-operator-4.1-configs
```
Comment (gavin-stackrox) [inline core-services/prow/02_config/_plugins.yaml:2247]:
```suggestion
      name: job-config-4.1
```

Comment (jschnath) [reply core-services/prow/02_config/_plugins.yaml]: Oh yeah, I messed that one up through duplication. Let me try with `job-config-4.1`

**Comment by jschnath** [comment]
To demonstrate @gavin-stackrox, this is how it fails https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/40216/pull-ci-openshift-release-master-correctly-sharded-config/1668498116355035136
without the __ocp in config. See code:
https://github.com/openshift/release/pull/40216/files#diff-a9035bbaee5af66130c81cfd0b97df589a7d2e22e87ad9d3800851407967c07b

So I'm readding, the __ocp, that seemed to fix this part at least.

**Comment by jschnath** [comment]
Hey colleagues, please approve this as we need it for the ACS 4.1 release

File: core-services/prow/02_config/_plugins.yaml [hunk 4]
```diff
@@ -1207,6 +1207,12 @@ config_updater:
         - ci
       gzip: true
       name: ci-operator-4.0-configs
+    ci-operator/config/stackrox/stackrox/*-release-4.1__ocp*.yaml:
+      clusters:
+        app.ci:
+        - ci
+      gzip: true
+      name: ci-operator-4.1-configs
```
[the remark this answers is not in the document]
Comment (jschnath) [reply core-services/prow/02_config/_plugins.yaml:1215]: I have not tried `name: ci-operator-misc-configs` should I?
The other one, `ci-operator/config/stackrox/stackrox/*-release-4.*__ocp*.yaml`, I have tried and the test failed.

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-4.1-presubmits.yaml [hunk 5]
```diff
@@ -0,0 +1,1488 @@
...
+      volumes:
+      - name: pull-secret
+        secret:
+          secretName: registry-pull-credentials
+      - name: result-aggregator
+        secret:
+          secretName: result-aggregator
+    trigger: (?m)^/test( | .* )gke-race-condition-qa-e2e-tests,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
[the remark this answers is not in the document]
Comment (jschnath) [reply ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-4.1-presubmits.yaml:258]: Oh, I will revisit that.

**Review by msugakov — APPROVED** [review]
/lgtm

**Pull request merged**